packages/core/src/error/index.ts

```diff
@@ -53,21 +54,26 @@ export interface IErrorDefinition {
  * It standardizes error reporting and can be extended for more specific error types.
  */
 export class MastraError extends Error {
-  public readonly id: string | number;
-  public readonly domain: `${Domain}`;
-  public readonly category: `${ErrorCategory}`;
+  public readonly id: Uppercase<string>;
+  public readonly domain: `${Domain | Uppercase<string>}`;
+  public readonly category: `${ErrorCategory | Uppercase<string>}`;
```

Comment:
They have to match teh domain and category
```suggestion
  public readonly domain: `${Domain}`;
  public readonly category: `${ErrorCategory}`;
```

Reply:
@wardpeet if we extend the MastraError class and wanted to create our own domain, for instance "MastraCloudBuild" shouldn't we keep it generic?

Otherwise we would have to update core everytime we would like to add a new domain